### PR TITLE
tests: Added unit tests for new key parts of Gamenative

### PR DESCRIPTION
**Pull request by unbelievableflavour**
## Description
Since I introduced a couple new key components in gamenative I thought it be good to include tests for those. New types for these should require to have tests at well so that at least these new sections are always reliable.

Tests for:
- Launch dependencies and its launch dep types.
- game fix registry and it's fix types.
- Preinstall steps and its specific preinstallstep types.

I've made the launchdeps, game fix registry, and preinstallsteps tests itself test independent of their children so they dont have to be updated on every new child types. (hence the small mock functions added to those so their lists can be mutated)

## Recording
https://discord.com/channels/1378308569287622737/1491337710244331720

## Checklist
- [x] If I have access to `#code-changes`, I have discussed this change there and it has been green-lighted. If I do not have access, I have still provided clear context in this PR. If I skip both, I accept that this change may face delays in review, may not be reviewed at all, or may be closed.
- [x] I have attached a recording of the change.
- [x] I have read and agree to the contribution guidelines in `CONTRIBUTING.md`.

## Summary by CodeRabbit

* **Tests**
  * Large new suite of unit and integration tests covering game fixes, dependency handling, launch-arg and registry behavior, pre-install steps, and orchestration across many edge cases.
* **Chores**
  * Added internal test-only hooks to allow injecting providers for fixes, dependencies, and pre-install steps to improve testability.
  * Minor cleanup: removed an unused import.

### CODE REVIEW

File: app/src/test/java/app/gamenative/utils/PreInstallStepsTest.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        override fun appliesTo(container: Container, gameSource: GameSource, gameDirPath: String): Boolean = applies
+
+        override fun buildCommand(
+            container: Container,
+            appId: String,
+            gameSource: GameSource,
+            gameDir: File,
+            gameDirPath: String,
+        ): String? = command
+    }
```
[the remark this answers is not in the document]
Comment (unbelievableflavour) [reply app/src/test/java/app/gamenative/utils/PreInstallStepsTest.kt:37]: Not true.
